Fix RTQ EQC test to model trimming correctly

Vagabond opened the pull request:
Fixed some silliness I was doing so that the test actually works right.

cc @metadave

Vagabond committed:
Fixup RTQ EQC test to work with pruning and parallel_commands

bookshelfdave commented on test/riak_repl2_rtq_eqc.erl:
```diff
@@ -40,10 +40,9 @@ rtq_test_() ->
 
 max_bytes() ->
-    ?LET(MaxBytes, nat(), (MaxBytes+1) * 512).
+    ?LET(MaxBytes, nat(), (MaxBytes+1) * 39).
```
Can you comment on why 39 was chosen?

Vagabond replied:
It is the size of the objects we are inserting into the queue (sans ETS overhead).

Vagabond committed:
[Bug-fix] Don't hand out a stale sequence number if a client re-registers

Instead, set the re-registered client to have its sequence id set to the
oldest entry in the queue. Not doing this can lead to the entire queue
crashing when it tries to ets:lookup a non-existant ID.

Vagabond commented:
Ok, this is actually ready to be reviewed now.

lordnull commented on test/riak_repl2_rtq_eqc.erl:
```diff
@@ -30,33 +31,84 @@
-                                                        ?MODULE:prop_main())))
+                                                        ?MODULE:prop_main()))),
+                ?assert(eqc:quickcheck(eqc:testing_time(25,
+                                                        ?MODULE:prop_parallel())))
```
Unloading the riak_repl_stats meck at this point prevents a large output of data due to cover being unable to find the riak_repl_stats_meck source file.

lordnull commented on test/riak_repl2_rtq_eqc.erl:
```diff
@@ -30,33 +31,84 @@
 max_bytes() ->
-    ?LET(MaxBytes, nat(), (MaxBytes+1) * 512).
+    ?LET(MaxBytes, nat(), {size, (MaxBytes+1) * 39}).
+
```
Why 39?

lordnull commented on test/riak_repl2_rtq_eqc.erl:
```diff
@@ -227,43 +312,44 @@ pull(#tc{name=Name}, Q) ->
+
+trim(Q, #state{max_bytes=Max}) ->
+    {_Size, NewQ} = lists:foldl(fun({Seq, NumItems, Bin}, {Size, Acc}) ->
+                case (39 + Size) > Max of
```
second time I've seen 39, so perhaps putting it in a descriptive macro?

lordnull committed:
Added teardown to eunit to avoid unneeded output

If a meck is not unloaded before the test process dies, and cover is
enabled, a large amount of output is printed. Cover is unable to find the
source file for the meck'ed module, and ends up printing the module raw
binary. Because this is not an error, it's best to simply silence it.

lordnull committed:
Replaced a magic number with a macro and comment.

lordnull commented:
Has my +1, but because it has some changes from me, I'll merge it once a 3rd party has given it a quick eyeball.

bookshelfdave commented:
looks good, we'll need to get this in 1.3
+1
